ls1intum/Artemis
# [Refactor] improved ExerciseResource

### Checklist

### Motivation and Context



### Description

applied new guidelines for Endpoints to ExerciseResource

### Steps for Testing


YOU NEED ADMIN TO TEST ALL ENDPOINTS IN THIS CLASS!
1. open the list of exercises for a Course
2. open the assessment dashboard of any course, then the exercise assessment dashboard of any type, check the breadcrumbs
3. look at the upcoming exercises of all courses (admin needed)
4. reset an exercise (by deleting all its participations, try to do it with a programming exercise)
5. cleanup a programming exercise (delete all build plans)
6. toggle the second correction round for a programming exercise


### Review Progress





- Code Review
- Manual Tests

### Test Coverage




ExerciseResource: 94%
